### ISSUE TITLE
MIDI lock contention

### ISSUE THREAD

**Issue by metalgearsloth** [space-wizards/RobustToolbox#4642]
ATM if I want the audio position + occlusion + velocity updates to be as frequent as standard audio this isn't possible because the thread and the parallel job will get lock contention with each other.

Ideally, at least on the update side:
- MIDI thread only locks per-renderer when it's using it rather than for every renderer.
- Whenever it needs to dispose of a renderer it just locks all of them then and does it / uses a semaphore or whatever
- Likewise the position etc update part of midimanager only locks that renderer.

**Comment by metalgearsloth** [comment]
The only thing that I changed in audio refactor was the update rate and how position / velocity are set for the buffered source so at least my best guess is lock contention, anything else is a new problem.

**Comment by deltanedas** [comment]
yeah maybe semaphore waiting will help with contention -> midi ddos goes bye bye????
